From 73b4a1dd62f8287c6e9e7e1d2f33bd1a31217358 Mon Sep 17 00:00:00 2001
From: Babu Shanmugam <bschanmu@redhat.com>
Date: Tue, 15 Nov 2016 11:42:12 +0000
Subject: [PATCH 1/3] OVN plugin configuration fixes

This patch renames certain ovn plugin and controller configuration
parameters as well as adds some additional ml2 configuration parameters.
It also disables the need for the neutron metadata agent.

Co-authored-by: Numan Siddique <nusiddiq@redhat.com>
Change-Id: Idc9e7ef4a1b88013bca3eac3c136e4710e38a5c0
---
 environments/neutron-ml2-ovn.yaml               |  8 +++++---
 puppet/services/neutron-compute-plugin-ovn.yaml | 10 +++++-----
 puppet/services/neutron-plugin-ml2-ovn.yaml     | 20 +++++++++++++++-----
 3 files changed, 25 insertions(+), 13 deletions(-)

diff --git a/environments/neutron-ml2-ovn.yaml b/environments/neutron-ml2-ovn.yaml
index 3da560c..4e70a50 100644
--- a/environments/neutron-ml2-ovn.yaml
+++ b/environments/neutron-ml2-ovn.yaml
@@ -3,6 +3,7 @@
 resource_registry:
   OS::TripleO::Services::NeutronL3Agent: OS::Heat::None
   OS::TripleO::Services::NeutronOvsAgent: OS::Heat::None
+  OS::TripleO::Services::NeutronMetadataAgent: OS::Heat::None
   OS::TripleO::Services::NeutronCorePlugin: OS::TripleO::Services::NeutronCorePluginML2OVN
   OS::TripleO::Services::ComputeNeutronCorePlugin: ../puppet/services/neutron-compute-plugin-ovn.yaml
 # Disabling Neutron services that overlap with OVN
@@ -12,11 +13,12 @@ resource_registry:
 
 parameter_defaults:
   NeutronMechanismDrivers: ovn
-  OVNSouthboundServerPort: 6642
-  OVNNorthboundServerPort: 6641
-  OVNDbConnectionTimeout: 60
   OVNVifType: ovs
   OVNNeutronSyncMode: log
   OVNQosDriver: ovn-qos
   OVNTunnelEncapType: geneve
   NeutronEnableDHCPAgent: false
+  NeutronTypeDrivers: 'geneve,vlan,flat'
+  NeutronNetworkType: 'geneve'
+  NeutronServicePlugins: 'qos,ovn-router'
+  NeutronVniRanges: ['1:65536', ]
diff --git a/puppet/services/neutron-compute-plugin-ovn.yaml b/puppet/services/neutron-compute-plugin-ovn.yaml
index ce28b5c..f825b48 100644
--- a/puppet/services/neutron-compute-plugin-ovn.yaml
+++ b/puppet/services/neutron-compute-plugin-ovn.yaml
@@ -18,9 +18,6 @@ parameters:
                  via parameter_defaults in the resource registry.  This
                  mapping overrides those in ServiceNetMapDefaults.
     type: json
-  OVNDbHost:
-    description: IP address on which the OVN DB servers are listening
-    type: string
   OVNSouthboundServerPort:
     description: Port of the Southbound DB Server
     type: number
@@ -37,9 +34,12 @@ outputs:
     value:
       service_name: neutron_compute_plugin_ovn
       config_settings:
-        tripleo::profile::base::neutron::agents::ovn::ovn_db_host: {get_param: OVNDbHost}
         ovn::southbound::port: {get_param: OVNSouthboundServerPort}
-        ovn::southbound::encap_type: {get_param: OVNTunnelEncapType}
+        ovn::controller::ovn_encap_type: {get_param: OVNTunnelEncapType}
         ovn::controller::ovn_encap_ip: {get_param: [ServiceNetMap, NeutronApiNetwork]}
+        tripleo.neutron_compute_plugin_ovn.firewall_rules:
+          '118 neutron geneve networks':
+            proto: 'udp'
+            dport: 6081
       step_config: |
         include ::tripleo::profile::base::neutron::agents::ovn
diff --git a/puppet/services/neutron-plugin-ml2-ovn.yaml b/puppet/services/neutron-plugin-ml2-ovn.yaml
index 59346ed..4d4c390 100644
--- a/puppet/services/neutron-plugin-ml2-ovn.yaml
+++ b/puppet/services/neutron-plugin-ml2-ovn.yaml
@@ -18,10 +18,14 @@ parameters:
     description: Mapping of service endpoint -> protocol. Typically set
                  via parameter_defaults in the resource registry.
     type: json
+  OVNSouthboundServerPort:
+    description: Port of the OVN Southbound DB server
+    type: number
+    default: 6642
   OVNDbConnectionTimeout:
     description: Timeout in seconds for the OVSDB connection transaction
     type: number
-    default: 60
+    default: 180
   OVNVifType:
     description: Type of VIF to be used for ports
     type: string
@@ -43,6 +47,10 @@ parameters:
     description: OVN notification driver for Neutron QOS service plugin
     type: string
     default: NULL
+  NeutronGeneveMaxHeaderSize:
+    description: Geneve encapsulation header size
+    type: number
+    default: 38
 
 resources:
 
@@ -61,10 +69,12 @@ outputs:
       config_settings:
         map_merge:
           - get_attr: [NeutronMl2Base, role_data, config_settings]
-          - neutron::plugins::ovn::ovsdb_connection_timeout: {get_param: OVNDbConnectionTimeout}
-            neutron::plugins::ovn::neutron_sync_mode: {get_param: OVNNeutronSyncMode}
-            neutron::plugins::ovn::ovn_l3_mode: true
-            neutron::plugins::ovn::vif_type: {get_param: OVNVifType}
+          - ovn::southbound::port: {get_param: OVNSouthboundServerPort}
+            neutron::plugins::ml2::ovn::ovsdb_connection_timeout: {get_param: OVNDbConnectionTimeout}
+            neutron::plugins::ml2::ovn::neutron_sync_mode: {get_param: OVNNeutronSyncMode}
+            neutron::plugins::ml2::ovn::ovn_l3_mode: true
+            neutron::plugins::ml2::ovn::vif_type: {get_param: OVNVifType}
             neutron::server::qos_notification_drivers: {get_param: OVNQosDriver}
+            neutron::plugins::ml2::max_header_size: {get_param: NeutronGeneveMaxHeaderSize}
       step_config: |
         include ::tripleo::profile::base::neutron::plugins::ml2
-- 
2.9.3

